Exercises/Caching/Cache.cs

```diff
@@ -19,28 +21,105 @@ public class Cache {
             //todo:// Checck if the cache is full. if true then remove the recently used item.
             if (IsCacheFull())  {
                 RemoveLeastRecentlyUsedItem();
+                AddItemToCache(key, value);
             } else  {
-                AddItemToCache();
+                AddItemToCache(key, value);
             }
             // todo:// else simply add item to cache.
         }
 
-        private void AddItemToCache() {
-            //todo://
+        private void AddItemToCache(string key, ImageDescription value) {
+            var node = doublyLinkedList.AddNewNodeAtHead(value, key);
+            dictionaryDoublyLinkedList.Add(key, node);
         }
 
         private bool IsCacheFull() {
             //todo://
-            return false;
+            return dictionaryDoublyLinkedList.Count == cacheSize;
         }
 
         private void RemoveLeastRecentlyUsedItem()  {
-            //todo://
+            var deletedNode = doublyLinkedList.DeleteNodeFromEnd();
+            dictionaryDoublyLinkedList.Remove(deletedNode.Data.Key);
         }
 
         public ImageDescription GetItem(string id)  {
             //todo:// if the item present return the item else return null.
+            if (dictionaryDoublyLinkedList.ContainsKey(id))
```

Comment:
This is the place where fetched item should be moved to head of the linked list:)